samples/Rules.Framework.WebUI.Sample/Rules/RulesRandomFactory.cs

```diff
@@ -17,75 +17,76 @@ public RulesRandomFactory()
-            Enums.ContentTypes.TestDecimal,
-            Enums.ContentTypes.TestLong,
-            Enums.ContentTypes.TestBoolean,
-            Enums.ContentTypes.TestShort,
-            Enums.ContentTypes.TestNumber,
-            Enums.ContentTypes.TestString,
-            Enums.ContentTypes.TestBlob,
+            RulesetNames.TestDateTime,
+            RulesetNames.TestDecimal,
+            RulesetNames.TestLong,
+            RulesetNames.TestBoolean,
+            RulesetNames.TestShort,
+            RulesetNames.TestNumber,
+            RulesetNames.TestString,
+            RulesetNames.TestBlob,
         };
 
         public IEnumerable<RuleSpecification> GetRulesSpecifications()
         {
             var currentYear = DateTime.UtcNow.Year;
             var rulesSpecifications = new List<RuleSpecification>();
 
-            foreach (var contentType in Enum.GetValues(typeof(ContentTypes)))
+            foreach (var ruleset in Enum.GetValues(typeof(RulesetNames)).Cast<RulesetNames>())
             {
                 for (var i = 1; i < random.Next(intialNumber, finalNumber); i++)
                 {
                     var dateBegin = CreateRandomDateBegin(currentYear);
 
-                    Add(CreateMultipleRule((ContentTypes)contentType, i, dateBegin, CreateRandomDateEnd(dateBegin)),
+                    Add(CreateMultipleRule(ruleset, i, dateBegin, CreateRandomDateEnd(dateBegin)),
                         RuleAddPriorityOption.ByPriorityNumber(i),
                         rulesSpecifications);
                 }
 
                 var deactiveDateBegin = CreateRandomDateBegin(currentYear);
 
-                Add(CreateMultipleRule((ContentTypes)contentType, finalNumber, deactiveDateBegin, CreateRandomDateEnd(deactiveDateBegin), isActive: false),
+                Add(CreateMultipleRule(ruleset, finalNumber, deactiveDateBegin, CreateRandomDateEnd(deactiveDateBegin), isActive: false),
                         RuleAddPriorityOption.ByPriorityNumber(finalNumber),
                         rulesSpecifications);
             }
 
             return rulesSpecifications;
         }
 
-        private static RuleBuilderResult<ContentTypes, ConditionTypes> CreateMultipleRule(
-            ContentTypes contentTypes,
+        private static RuleBuilderResult<RulesetNames, ConditionNames> CreateMultipleRule(
+            RulesetNames ruleset,
             int value,
             DateTime dateBegin,
             DateTime? dateEnd,
-            bool isActive = true) => Rule.New<ContentTypes, ConditionTypes>()
-                .WithName($"Multi rule for test {contentTypes} {value}")
-                .WithContent(contentTypes, new { Value = value })
-                .WithDatesInterval(dateBegin, dateEnd)
+            bool isActive = true) => Rule.Create<RulesetNames, ConditionNames>($"Multi rule for test {ruleset} {value}")
+                .OnRuleset(ruleset)
+                .SetContent(new { Value = value })
```

Comment:
There is a race condition in this method if a thread changes the object that was created, on the next match the object changed will be given to the new thread. Perhaps we can allow the dev to have a function to be able to give the thread a new object whenever the rule matches.

Reply:
I agree, but I will address that on a separate PR 👍